core/common/src/-Util.kt

```diff
@@ -19,17 +19,28 @@
  * limitations under the License.
  */
 
-package kotlinx.io
+@file:Suppress("NOTHING_TO_INLINE")
 
-import kotlin.native.concurrent.SharedImmutable
+package kotlinx.io
 
-@SharedImmutable
 internal val HEX_DIGIT_CHARS =
   charArrayOf('0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f')
 
 internal fun checkOffsetAndCount(size: Long, offset: Long, byteCount: Long) {
-  if (offset or byteCount < 0 || offset > size || size - offset < byteCount) {
-    throw ArrayIndexOutOfBoundsException("size=$size offset=$offset byteCount=$byteCount")
+  if (offset < 0 || offset > size || size - offset < byteCount || byteCount < 0) {
+    throw IllegalArgumentException("offset: $offset, byteCount: $byteCount, size: $size")
```

Comment:
According to this message, it is not entirely clear what exactly the error is, without knowing the implementation of the method.

Perhaps it is worth at least specifying in the message how these values should be related?

Reply:
Sure, I'll add more context to the message.